src/js/popup/popup.js

```diff
@@ -26,6 +26,19 @@
 
         // Custom rule to send "Closed Report Issue" event
         if (e.target.dataset.navId && e.target.dataset.navId === "webcompat") {
+
+          // Reset Form
+          popup.panel.webcompat.resetForm();
+
+          // Remove event listeners
+          const panel = document.getElementById('webcompat-panel');
+          const clonedPanel = panel.cloneNode(true);
+          panel.parentNode.replaceChild(clonedPanel, panel);
+
+          // Set back button listeners again for webcompat
+          const webcompatBackButton = document.querySelector("#webcompat-panel .fx-relay-panel-header-btn-back");
+          webcompatBackButton.addEventListener("click", popup.events.backClick, false);
```

Comment:
Because we return the user back to the settings page after submitting the form, we have to do a heavy reset of the form so they can fill it out again. 

The `replaceChild(clonedPanel, panel);` part is a [hacky way](https://stackoverflow.com/questions/65495153/remove-all-event-listeners-from-an-element-javascript) to remove all anonymous event listeners on an element.

Reply:
Not sure if this helps since you already have a working way, but couldn't we also do something like: 

```
// first we get ref. to element in question 
const element = document.getElementById('webcompat-panel');

// call getEventListeners to get an object representing all the event listeners on the element
const listeners = getEventListeners(element);

for (const eventType in listeners) {
  const eventListeners = listeners[eventType];

// iterate through each event type (e.g. 'click', 'mouseenter', etc.) in the listeners object, 
// and for each event type we iterate through the array of event listeners for that type. 
  for (const eventListener of eventListeners) {

    // If an event listener doesn't have a name property (which means it's an anonymous function)
    if (!eventListener.name) {

      // we remove it 
      element.removeEventListener(eventType, eventListener.listener);
    }
  }
}```